src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -124,22 +125,39 @@ protected void PopulateProperties(JsonSerializer serializer, object obj, JsonRea
             }
         }
 
-
         public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
         {
             if (DocumentRootConverter.TryResolveAsRootData(writer, value, serializer))
+            {
                 return;
+            }
+
+            // if they have custom convertors registered, we will respect them
+            JsonConverter customConvertor = null;
 
-            //if they have custom convertors registered, we will respect them
-            var customConvertor = serializer.Converters.FirstOrDefault(x => x.CanWrite && x.CanConvert(value.GetType()));
-            if (customConvertor != null && customConvertor != this)

```

Comment:
You have droped the `customConvertor != this` check.

This was needed at some point (but was possibly made redundant on the last refactor). If it didnt break test im guessing it should be fine